core/src/main/java/org/ehcache/core/config/ResourcePoolsImpl.java

```diff
@@ -61,34 +62,29 @@ public ResourcePool getPoolForResource(ResourceType resourceType) {
    */
   @Override
   public ResourcePools validateAndMerge(ResourcePools toBeUpdated) {
+    // Ensure update pool types already exist in existing pools
     if(!getResourceTypeSet().containsAll(toBeUpdated.getResourceTypeSet())) {
       throw new IllegalArgumentException("Pools to be updated cannot contain previously undefined resources pools");
     }
+    // Can not update OFFHEAP
+    // TODO: Move to OFFHEAP ResourcePool
```

Comment:
I'm converting the `TODO`s to issues for all except the `ClusteredStore` class since this class still needs major work.